src/config/mod.rs

```diff
@@ -466,10 +470,17 @@ pub struct TransformOuter {
     pub inner: Box<dyn TransformConfig>,
 }
 
+pub type EnrichmentTableList =
+    Arc<RwLock<HashMap<String, Box<dyn enrichment_tables::EnrichmentTable + Send + Sync>>>>;
+
 #[async_trait]
 #[typetag::serde(tag = "type")]
 pub trait TransformConfig: core::fmt::Debug + Send + Sync + dyn_clone::DynClone {
-    async fn build(&self, globals: &GlobalOptions) -> crate::Result<transforms::Transform>;
+    async fn build(
+        &self,
+        enrichment_tables: EnrichmentTableList,
```

Comment:
I don't think I would add this to `build` directly. It seems better suited to either tacking onto `GlobalOptions` or combining with `GlobalOptions` into a `TransformContext` or similar.

Reply:
Adding it to `GlobalOptions` doesn't feel very elegant since the `GlobalOptions` are deserialized separately to the enrichment_table sections. A simple `TransformContext` seems good.